### DISCUSSION TITLE
HTML templates not working v0.49.1

### DISCUSSION THREAD

**Discussion by ryan-cpi** [aquasecurity/trivy#6316] [Q&A] [answered]
### Question

I want to have my output formatted into an HTML file. The [docs here](https://aquasecurity.github.io/trivy/v0.17.2/examples/report/#html) say to run `trivy image --format template --template "@contrib/html.tpl" -o report.html golang:1.12-alpine` however, the resulting file only shows the text "html.tpl" and shows no results.

When running locally I downloaded and pointed to this template `html.tpl` taken from [here](https://github.com/aquasecurity/trivy/blob/main/contrib/html.tpl)

Is there a reason why `trivy config . --format template --template "html.tpl" -o report.html` would fail to show findings when `trivy config . --format json` outputs results? It seems like the HTML file provided is out of date or no longer works with the latest version?

Running with `--debug` shows its finding results just not outputting them to the file. `2024-03-13T14:40:36.723Z DEBUG   [misconf] 40:36.723172000 terraform.executor`

### Target

None

### Scanner

Misconfiguration

### Output Format

Template

### Mode

Standalone

### Operating System

macOS Sonoma 14.3.1 

### Version

```bash
brew install trivy
trivy --version
Version: 0.49.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-03-13 12:12:01.708119317 +0000 UTC
  NextUpdate: 2024-03-13 18:12:01.708118946 +0000 UTC
  DownloadedAt: 2024-03-13 13:07:42.318308 +0000 UTC
Policy Bundle:
  Digest: sha256:cdff1bc8c97e4f5cd04782b057c00f5ea8cd81147a506ac4be76bef13710f2d3
  DownloadedAt: 2024-03-13 13:04:14.157532 +0000 UTC
```

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @ryan-cpi 
Thanks for your interest to Trivy!

You need to add `@` as prefix for template path.
e.g.:
```
➜ tree
.
└── contrib
    └── html.tpl

➜ trivy image --format template --template "@contrib/html.tpl" -o report.html golang:1.12-alpine

➜  cd contrib

➜  contrib ls -hl
total 16
-rw-r--r--  1 work  staff   4.9K Oct 31 12:24 html.tpl

➜  trivy image --format template --template "@html.tpl" -o report.html golang:1.12-alpine 
```

Same for misconfigs - `➜ trivy config --format template --template "@contrib/html.tpl" -o report-config.html ./Dockerfile`:
![изображение](https://github.com/aquasecurity/trivy/assets/91113035/5953bd40-b269-4a4f-a15e-f9f7aaa80741)

If you need to see all checks - use `--include-non-failures` flag.

Regards, Dmitriy

**Reply by ryan-cpi** [reply]
Thanks for clarifying that for me. Do you know why adding a @ before the file name is needed here its not something I've seen others follow.
I was able to generate some output to the html.tpl file but when running for Terraform there's nothing linking the results to files, folders or lines. Is this something that's planned to change in the future? Our current IAC scanner kics has this functionality then we create an HTML file from the JSON output with all the GitLab and external links and descriptions needed.

![image](https://github.com/aquasecurity/trivy/assets/90183686/e589e201-bc3c-4f9b-add7-ff71233aa22d)

**Reply by DmitriyLewen** [reply]
> when running for Terraform there's nothing linking the results to files, folders or lines

@nikpivkin @simar7 can you take a look?

**Reply by simar7** [reply]
At the moment the we don't have support for linking results to files as you mentioned but it is something you could write with the help of [trivy output plugins](https://aquasecurity.github.io/trivy/v0.49/docs/advanced/plugins/).

**Reply by ryan-cpi** [reply]
Thanks for letting me know. I've accepted the previous answer. Is there an open issue for this or is it planned to be added sometime? 
This issue prevents us from replacing the current scanner we use in Gitlab kics. Without links taking the user to the line in question or with no details on what terraform folder it's running against the output isn't too usable for our us in this current state.
